src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,43 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  currentTodo: Todo | null,
+  setCurrentTodoId: (id: number | null) => void,
+};
+
+export const TodoModal: React.FC<Props> = ({
+  currentTodo,
+  setCurrentTodoId,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    if (currentTodo) {
+      getUser(currentTodo.userId).then(currentUser => {
+        setUser(currentUser);
+      });
+    }
+  }, []);
+
+  if (!currentTodo) {
+    return null;
+  }
+
+  const {
+    completed,
+    id,
+    title,
+  } = currentTodo;
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {!user ? (
```

Comment:
Let's create a boolean useState `isLoading` and use it here.
Same as you did in `app.tsx`